### DISCUSSION TITLE
Fractional Exponents in pow()

### DISCUSSION THREAD

**Discussion by rontarrant** [godotengine/godot-proposals#5087] [Scripting]
I think it would be handy to have fractional exponents in GDScript's pow() method. Python allows them; C# does, too. And even though we can mix C# with GDScript to bring the functionality of fractional exponents to the latter, it would be nice to use pure GDScript instead.

I know I'm not making a very strong argument for this, but maybe it's not such a big deal to implement?

**Comment by miv391** [comment]
You already can use floats as exponent. `pow(9, 0.5)` returns 3.

**Comment by rontarrant** [comment]
It doesn't work with an exponent of (1 / 12) or the decimal equivalent
thereof. I'm guessing this is related to the fact that Godot uses
single-precision floating point numbers (as opposed to double-precision).

**Reply by Calinou** [reply]
Double-precision floats can be enabled with the `float=64` SCons option in 4.0.alpha, but this requires compiling your own editor and export template binaries. This is also not available in 3.x.

**Reply by miv391** [reply]
Do you have an example which returns an incorrect value? I tried a few powers of 1/12 with Godot v3.4.4 and got the same results as my old calculator does.

**Reply by rontarrant** [reply]
I think I figured it out: mixed types. If an int is tossed into an equation with a float, the result seems to be downgraded to an int. I haven't 100% verified this yet, but for the mortgage calculator I'm working on, it does hold true.
